libosmscout-client-qt/src/osmscout/ElevationChartWidget.cpp

```diff
@@ -52,6 +53,98 @@ ElevationChartWidget::~ElevationChartWidget()
   }
 }
 
+namespace {
+class DistanceUnit {
```

Comment:
In base library we have Distance to hold distance (obviously). Then enum class `osmscout::Units` with Metrics and Imperial enumerators. Then few types (structs Meter, Kilometer, Mile, Yard, Feet) for usage in template methods of Distance...

But here I am unable to decide in compile time. I needed some type that represents distance unit and may be overloaded and changed at runtime. In future it would be great to have one unit representation in base library that may be used in compile-time, as template argument and at runtime... It will not be complicated, but I would do it in separate merge request if you agree...?